src/ColorVectorSpace.jl

```diff
@@ -94,6 +94,21 @@ divtype(::Type{A}, ::Type{B}) where {A,B<:FixedPoint} = coltype(typeof(zero(A)/t
 powtype(::Type{A}, ::Type{B}) where {A,B} = coltype(typeof(zero(A)^zero(B)))
 sumtype(a::Colorant, b::Colorant) = coltype(sumtype(eltype(a),eltype(b)))
 
+multype(::Type{Bool}, ::Type{B}) where {B} = B
+multype(::Type{A}, ::Type{Bool}) where {A} = A <: Integer ? multype(A, N0f8) : A
+multype(::Type{Bool}, ::Type{Bool}) = Bool
+sumtype(::Type{Bool}, ::Type{B}) where {B} = B <: Integer ? N0f8 : sumtype(N0f8, B) # FIXME
+sumtype(::Type{A}, ::Type{Bool}) where {A} = sumtype(Bool, A)
+sumtype(::Type{Bool}, ::Type{Bool}) = N0f8 # FIXME
+divtype(::Type{Bool}, ::Type{B}) where {B} = divtype(B <: Integer ? N0f8 : UInt8, B)
```

Comment:
No, but I think it's perfectly fine to evaluate both the true and false cases here as they're constant DataType.

> From `?ifelse`:
> In some cases, using `ifelse` instead of an if statement can eliminate the branch in generated code and provide
  higher performance in tight loops.

The `@code_native` looks the same for this case so I don't know if it worths. Maybe Julia is just smart enough to optimize it...

Reply:
Even using `ifelse` does not completely suppress the branches, so I simply prefer to keep it short and readable.
In the first place, since this is a matter of constant propagation, I don't think it has anything to do with the branch generation.